src/server/network.ts

```diff
@@ -275,15 +294,33 @@ export class Response extends SdkObject {
   private _headersMap = new Map<string, string>();
   private _getResponseBodyCallback: GetResponseBodyCallback;
   private _timing: ResourceTiming;
+  private _connectionDetails: ConnectionDetails;
+  private _connectionDetailsFinishedCallback: (details: ConnectionDetails) => void = _ => {};
+  private _connectionDetailsPromise: Promise<ConnectionDetails>;
 
-  constructor(request: Request, status: number, statusText: string, headers: types.HeadersArray, timing: ResourceTiming, getResponseBodyCallback: GetResponseBodyCallback) {
+  constructor(request: Request, status: number, statusText: string, headers: types.HeadersArray, timing: ResourceTiming, getResponseBodyCallback: GetResponseBodyCallback, connectionDetails: ConnectionDetails & {complete?: boolean}) {
```

Comment:
Let's not pass in the connection details here, and only pass them in fully formed to _requestFinished.